backup: TestOnlineRestoreDistFlowSplitScatter failed

backup.TestOnlineRestoreDistFlowSplitScatter [failed](https://mesolite.cluster.engflow.com/invocations/default/f1d46fba-9660-44fa-b212-15f76f994873?testReportRun=2&testReportShard=48&testReportAttempt=1#targets-Ly9wa2cvYmFja3VwOmJhY2t1cF90ZXN0) on master @ [5e1ab57f0485ead6fb19461bad9e954a148efe19](https://github.com/cockroachdb/cockroach/commits/5e1ab57f0485ead6fb19461bad9e954a148efe19):

Fatal error:

```
panic: write-job-message-insert: failed to read query result: query execution canceled

```
Stack: 

```
goroutine 102581 [running]:
github.com/cockroachdb/cockroach/pkg/util/besteffort.Warning({0x75d2050, 0x13324bc3ad80}, {0x70bbcc0, 0x16}, 0x133220bb6518)
	pkg/util/besteffort/besteffort.go:34 +0x1fa
github.com/cockroachdb/cockroach/pkg/backup.(*restoreResumer).doDownloadFilesWithRetry(0x13324a9a0b00, {0x75d2050, 0x13324bc3ad80}, {0x7640970, 0x13323bb7d5d0})
	pkg/backup/restore_online.go:900 +0x5c5
github.com/cockroachdb/cockroach/pkg/backup.(*restoreResumer).doResume(0x13324a9a0b00, {0x75d2050, 0x13324bc3ad80}, {0x6cad080?, 0x13323bb7d5d0})
	pkg/backup/restore_job.go:2229 +0x325
github.com/cockroachdb/cockroach/pkg/backup.(*restoreResumer).Resume(0x13324a9a0b00?, {0x75d2050?, 0x13324bc3ad80?}, {0x6cad080?, 0x13323bb7d5d0})
	pkg/backup/restore_job.go:2209 +0xb8
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine.func2(0x133237329380?, 0x0?, 0x133220bb78c0, {0x75a8b00?, 0x13324a9a0b00?}, {0x75d2050?, 0x13324bc3ad80?}, {0x6cad080?, 0x13323bb7d5d0?})
	pkg/jobs/registry.go:1661 +0xb4
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine(0x13324a26a008, {0x75d44f8, 0x1332316655e0}, {0x6cad080, 0x13323bb7d5d0}, {0x75a8b00, 0x13324a9a0b00}, 0x133237329380, {0x133251228ac2, 0x7}, ...)
	pkg/jobs/registry.go:1662 +0xb7f
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).runJob(0x13324a26a008, {0x75d44f8, 0x1332316655e0}, {0x75a8b00, 0x13324a9a0b00}, 0x133237329380, {0x133251228ac2, 0x7}, {0x133241583530, 0x17})
	pkg/jobs/adopt.go:503 +0x585
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resumeJob.func1({0x75d44f8?, 0x133231665360?})
	pkg/jobs/adopt.go:350 +0x13c
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func1({0x75d44f8, 0x133231665360})
	pkg/util/stop/stopper.go:447 +0x8a
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 102560
	pkg/util/stop/stopper.go:445 +0x125
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestOnlineRestoreDistFlowSplitScatter
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestOnlineRestoreDistFlowSplitScatter2269326138
    test_log_scope.go:82: use -show-logs to present logs inline
=== RUN   TestOnlineRestoreDistFlowSplitScatter/goroutine-path
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=2</code>
 - <code>shard=48</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestOnlineRestoreDistFlowSplitScatter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-63133

--- a/pkg/backup/restore_online_distflow_test.go
+++ b/pkg/backup/restore_online_distflow_test.go
@@ -52,6 +52,8 @@ func TestOnlineRestoreDistFlowSplitScatter(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
+	AllowORDownloadBestEffortFailures(t)
+
 	// Block reads of backup data SSTs to simulate high S3 latency.
 	blockCh := make(chan struct{})
 	defer nodelocal.ReplaceNodeLocalForTestingWithInterceptor(

